torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1889,11 +1889,15 @@ XLATensor XLATensor::sum(const XLATensor& input,
                          std::vector<xla::int64> dimensions,
                          bool keep_reduced_dimensions,
                          c10::optional<at::ScalarType> dtype) {
+  if (at::isIntegralType(input.dtype()) && !dtype) {
+    dtype = at::ScalarType::Long;
```

Comment:
Hmm, why not letting it get input.dtype instead of bolting in kLong?

Reply:
Because that the PyTorch semantics. If the input type is any sort of integral, the output type is a long.

Reply:
So int+int == long?

Reply:
Yes.